compiler/src/iree/compiler/Dialect/Stream/Transforms/test/refine_usage.mlir

```diff
@@ -210,3 +210,75 @@ func.func @escapingAlloca() -> !hal.buffer_view {
   %2 = stream.tensor.export %1 : tensor<f32> in !stream.resource<external>{%c123} -> !hal.buffer_view
   return %2 : !hal.buffer_view
 }
+
+// -----
+
+// CHECK-LABEL: @testIf
+func.func @testIf(%arg0: i1, %arg1: !hal.buffer_view, %arg2: !hal.buffer_view) -> !hal.buffer_view attributes {iree.abi.stub} {
+  %c4 = arith.constant 4 : index
+  %c0 = arith.constant 0 : index
+  %c268435488_i32 = arith.constant 268435488 : i32
+  %c1_i32 = arith.constant 1 : i32
+  %c1 = arith.constant 1 : index
+  hal.buffer_view.assert<%arg1 : !hal.buffer_view> message("input 1") shape([%c1]) type(%c268435488_i32) encoding(%c1_i32)
+  %0 = stream.tensor.import %arg1 : !hal.buffer_view -> tensor<1xi32> in !stream.resource<external>{%c4}
+  %1 = stream.async.transfer %0 : !stream.resource<external>{%c4} -> !stream.resource<*>{%c4}
+  hal.buffer_view.assert<%arg2 : !hal.buffer_view> message("input 2") shape([%c1]) type(%c268435488_i32) encoding(%c1_i32)
+  %2 = stream.tensor.import %arg2 : !hal.buffer_view -> tensor<1xi32> in !stream.resource<external>{%c4}
+  %3 = stream.async.transfer %2 : !stream.resource<external>{%c4} -> !stream.resource<*>{%c4}
+  // CHECK: %[[IF:.+]] = scf.if
+  // CHECK-SAME: !stream.resource<external>
+  %4 = scf.if %arg0 -> (!stream.resource<*>) {
+    // CHECK: %[[DISPATCH:.+]] = stream.async.dispatch
+    // CHECK-SAME: !stream.resource<external>
+    // CHECK-SAME: !stream.resource<external>
+    // CHECK-SAME: -> !stream.resource<external>
+    %7 = stream.async.dispatch @testIf_dispatch_0::@testIf_dispatch_0_generic(%1[%c0 to %c4 for %c4], %3[%c0 to %c4 for %c4]) : (!stream.resource<*>{%c4}, !stream.resource<*>{%c4}) -> !stream.resource<*>{%c4}
```

Comment:
the names don't matter here and can be kept short
```suggestion
    %7 = stream.async.dispatch @ex::@dispatch(%1[%c0 to %c4 for %c4], %3[%c0 to %c4 for %c4]) : (!stream.resource<*>{%c4}, !stream.resource<*>{%c4}) -> !stream.resource<*>{%c4}
```

Reply:
Fixed